### PR TITLE
Improve test workflow on CI

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Ref: https://github.com/stylelint/.github/pull/18

> Is there anything in the PR that needs further explanation?

- Split a test job for coverage from matrix
- Use npm 9 (latest)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,7 +45,8 @@
     "prepare": "husky install && patch-package",
     "release": "np --no-release-draft",
     "pretest": "npm run lint",
-    "test": "jest --coverage",
+    "test": "jest",
+    "test:coverage": "jest --coverage",
```
Comment (jeddy3) [inline package.json:49]:
```suggestion
    "test-coverage": "jest --coverage",
```

**Comment by ybiquitous** [comment]
Umm..., why do triggers not work?

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -9,50 +9,42 @@ on:
...
         run: npm ci
 
-      - name: Test
-        run: npm run jest
-        if: "!(startsWith(matrix.os, 'ubuntu') && matrix.node == 18)"
-
       - name: Test with coverage
-        run: npm run jest -- --coverage
-        if: startsWith(matrix.os, 'ubuntu') && matrix.node == 18
+        run: npm run test:coverage
```
Comment (jeddy3) [inline .github/workflows/testing.yml:43]:
```suggestion
        run: npm run test-coverage
```

Minor nit as we use `:` to signify a subtask that can be run in parallel, e.g. `lint:*`.

**Review by jeddy3 — CHANGES_REQUESTED** [review]
Thank you.

I've requested minor changes.

I appreciate all these refinements you put into our workflows. It makes it easier for us to consistently maintain all the Stylelint repos.

File: .github/workflows/testing.yml (hunk 2 above)
Comment (ybiquitous) [reply .github/workflows/testing.yml]: Make sense. Fixed via 3827f4be.

**Comment by ybiquitous** [comment]
All jobs in the matrix and a job with coverage work well!

**Review by jeddy3 — APPROVED** [review]
Nice! LGTM, thank you.

**Pull request merged**